### ISSUE TITLE
Angular memory issue

### ISSUE THREAD

**Issue by akash-wadhwa** [angular/angular#8946]
- **I'm submitting a ...**
  bug

**Current behavior**
I am starting a project where most of the information will be  available on single page with horizontal scroll. I was doing load testing with angular default "tour of heroes" app. I added around 800 input elements with two way data binding in "dashboard.component". I started adding text in one of the input boxes and I was constantly looking at the memory consumption. It gradually rises to 3GB as I was adding the text and browser crashed. Is 800 is the max limit for the input elements with two way binding in angula2 ?

**Expected/desired behavior**
I was expecting a much higher limit
- **What is the motivation / use case for changing the behavior?**
  With Angular2 expectation was much higher for building large scale applications.
- **Please tell us about your environment:**
  Windows 8.1
- **Angular version:** 
  Angular 2.0.0-rc.1
- **Browser:**
  Chrome
- **Language:**
  TypeScript

**Comment by chandandas** [comment]
I am also facing same issues. 

Can anyone provide solution to it.

**Comment by zoechi** [comment]
Please add some Plunker to reproduce.

**Comment by thelgevold** [comment]
The limit will vary based on hardware, but in any event, 800 input fields on the same page seems impractical. If you want to a more scalable solution I would recommend using a virtualized UI where you only bind UI elements to a fixed window into the larger data set. Typically based on what's visible on the screen. 

If you only generate DOM for a fixed window into the DOM you will have a very scalable UI

**Comment by robwormald** [comment]
We'll need a reproduction, as stated in the issue template, to investigate this.

**Comment by kylecordes** [comment]
@robwormald  I made the following a few days ago to stress things in a different direction, generating a (configurable) pile of components and nesting them together automatically. Would something similar help here, a way to emit a stress-test project with a configurable number of controls per component?

https://www.npmjs.com/package/angular2-stress-test

**Comment by vicb** [comment]
Has it only been tested on window ? which browsers ?

**Comment by tbosch** [comment]
Is the problem here that the memory is not being freed, or that we are just using too much?

**Comment by DzmitryShylovich** [comment]
no activity, no repro. close?

**Comment by kylecordes** [comment]
@DzmitryShylovich and others, I took a few minutes to create a simple case of using many form controls in a Angular application. Code here:

https://github.com/kylecordes/large-angular-form-test


I'm running on MacOS 10.12.1, Retina MBP. Chrome 55.0.2883.87 (64-bit).

First I created a fresh new project, using CLI. Unfortunately that means there's a bunch of CLI complexity in the way of testing the effect of project size on billed tooling resource consumption. But this item is only about Angular itself, and I think the results of the running Angular app should be about the same regardless of the details of the build tooling.

To measure consumption in Chrome, I started with a fresh tab each time. This seemed to yield a repeatable result, while automatic reloading of new code in the same tab showed a steadily increasing result which would invalidate the measurements here.


Initial commit, as created by CLI beta.21:

Chrome task manager: 94 MB


With 500 fields:

Chrome task manager: 184 MB


It seems to me that 90 MB is quite a bit to add just to have 500 text fields on the screen. But I have no sense (and did not attempt to test) how much of this was from Chrome versus Angular.

Someone interested in digging in further (probably worthwhile only if the core team considers these numbers big enough to potentially take action on) could investigate the extent to which just adding the dominant elements with no data binding consumes memory.
